proposals/1859_github-auth/README.md

```diff
@@ -0,0 +1,527 @@
+    # the group name in Kubernetes. Can be either "name" or "slug".
+    # Team names can contain upper and lower case characters,
+    # whitespace, and punctuation (e.g. "Kube admins!").
+    # Slug names are lower case alphanumeric characters and may
+    # contain dashes (e.g. "kube-admins").
+    # Either way, group names will always be prefixed by the
+    # GitHub org name followed by a colon (e.g. my-org:my-team).
+    # Note that colons are not allowed in GitHub org names,
+    # so the first colon in the group name will always be the
+    # org/team separator colon.
+    # If desired, an admin could configure identity
+    # transformation expressions on a FederationDomain to
+    # further customize these group names.
+    # Defaults to "slug".
+    groups: slug
+  # Optionally reject any user who attempts to authenticate
+  # unless they belong to one of these GitHub orgs.
+  # The GitHub App or GitHub OAuth App provided in the
+  # spec.client.secretName must be allowed by the org
+  # owners to view org memberships, or else the
+  # user will not be considered to belong to that org
+  # or belong to any teams within that org.
+  # When specified, users' group memberships will be
+  # filtered to include only those GitHub teams which
+  # are owned by these orgs.
+  # If desired, an admin could configure identity
+  # policy expressions on a FederationDomain to
+  # further customize which users and groups are allowed
+  # to authenticate.
+  # When not set, any GitHub user can authenticate
+  # regardless of org membership, and all the team
+  # memberships returned by the GitHub API for that
+  # user will be reflected as groups regardless of
+  # which orgs own those teams.
+  allowOrganizations: [ vmware-tanzu, broadcom ]
+  # Allow or disallow users to use GitHub Personal Access Tokens
+  # for CLI-based (no web browser) authentication.
+  # Setting these both to false disables all non-interactive
+  # CLI-based auth, but still allows browser-based auth.
+  allowPersonalAccessTokens:
+    # Allow or disallow the use of GitHub Fine-grained PATs
+    # to authenticate to Kubernetes clusters. If they
+    # are disallowed for your GitHub org, then also
+    # disallow them here so your users can get a nice
+    # error message when trying to use a fine-grained PAT.
+    # Defaults to false.
+    fineGrained: true
+    # Allow or disallow the use of GitHub Classic PATs
+    # to authenticate to Kubernetes clusters. If they
+    # are disallowed for your GitHub org, then also
+    # disallow them here so your users can get a nice
+    # error message when trying to use a classic PAT.
+    # Defaults to false.
+    classic: false
+  client:
+    # The name of Secret in the same namespace which holds the
+    # client ID and client secret of a GitHub App or
+    # Github OAuth App. This will only be used for web
+    # browser-based auth flows (not CLI-based flows).
+    # Required.
+    secretName: github-client-credentials
```

Comment:
It may be useful to show the format of this secret somewhere in this proposal as well (for clarification re: `data.clientID, data.clientSecret`, or whatever the contents).  I'm assuming it will be effectively this:

```yaml
apiVersion: v1
type: secrets.pinniped.dev/github-client # per this in the first draft: https://github.com/vmware-tanzu/pinniped/pull/1900/files#diff-ebf3585244ef6bf940ef116dee8ea43c7cbbfa525637a94977fc576ccd096a52R160
kind: Secret
metadata:
  name: github-client-credentials
data:
  clientID: foobar
  clientSecret: baz
```

Reply:
TODO: add this to the proposal